### PR TITLE
Add JSDoc types for ContextExclusionPlugin

### PR DESCRIPTION
**Pull request by mohsen1**

### CODE REVIEW

File: lib/ContextExclusionPlugin.js [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
 
+	/**
+	 * Apply the plugin
+	 * @param {Compiler} compiler Webpack Compiler
+	 * @returns {void}
+	 */
 	apply(compiler) {
-		compiler.hooks.contextModuleFactory.tap("ContextExclusionPlugin", cmf => {
+		compiler.hooks.contextModuleFactory.tap("ContextExclusionPlugin", (
+			/** @type {ContextModuleFactory} */ cmf
```
Comment (ooflorent) [inline lib/ContextExclusionPlugin.js]: Why is it required? Should't `compiler.hooks.contextModuleFactory` be typed?

Comment (mohsen1) [reply lib/ContextExclusionPlugin.js]: Good question. Let me check

Comment (sokra) [reply lib/ContextExclusionPlugin.js]: Should not be needed

**Comment by sokra** [comment]
Thanks

**Pull request merged**